[DOC] Add sort processor documentation (#5990)
opensearch-project/documentation-website: _ingest-pipelines/processors/sort.md

---
layout: default
title: Sort
parent: Ingest processors
nav_order: 250
---

# Sort processor

> The `sort` processor sorts an array of items in either ascending or descending order. The processor takes in a field name (`field`) and a sort order (`order`), and it creates a new field (`target_field`) with the sorted array.

Author: ```suggestion
The `sort` processor sorts an array of items in either ascending or descending order. Numeric arrays are sorted numerically, while string or mixed arrays (strings and numbers) are sorted lexicographically. Throws an error if the input is not an array.
```

Author: @gaobinlong Please confirm this change addresses your comments accurately. Thank you.

> The `sort` processor sorts an array of items in either ascending or descending order. Numeric arrays are sorted numerically, while string or mixed arrays (strings and numbers) are sorted lexicographically. Throws an error if the input is not an array.

Reviewer: ```suggestion
The `sort` processor sorts an array of items in either ascending or descending order. Numeric arrays are sorted numerically, while string or mixed arrays (strings and numbers) are sorted lexicographically. The processor throws an error if the input is not an array.
```

The following is the syntax for the `sort` processor:

```json
{
  "description": "Sort an array of items",
  "processors": [
    {
      "sort": {
        "field": "my_array_field",
        "order": "desc"
      }
    }
  ]
}
```
{% include copy-curl.html %}

## Configuration parameters

The following table lists the required and optional parameters for the `sort` processor.

| Parameter  | Required/Optional  | Description  |
|---|---|---|

> `field`  | Required | The field to be sorted.

Author: ```suggestion
`field`  | Required | The field to be sorted.
```
```suggestion
`field`  | Required | The field to be sorted. Must be an array.
```

> `order`  | Required | The sort order to apply. Accepts `asc` for ascending or `desc` for descending. Default is `asc`.

Author: ```suggestion
`order`  | Optional | The sort order to apply. Accepts `asc` for ascending or `desc` for descending. Default is `asc`.
```

> `target_field` | Optional | The name of the field where the sorted array is stored. If not specified, the sorted array is stored in the same field as the original array (the `field` variable). 

Reviewer: ```suggestion
`target_field` | Optional | The name of the field in which the sorted array is stored. If not specified, then the sorted array is stored in the same field as the original array (the `field` variable). 
```

```json
POST _ingest/pipeline/sort-pipeline/_simulate
{
  "docs": [
    {
      "_source": {
        "my_array_field": [3, 1, 4, 1, 5, 9, 2, 6, 5]
      }
    }
  ]
}
```
{% include copy-curl.html %}

#### Response

The following example response confirms that the pipeline is working as expected:

```json
{
  "docs": [
    {
      "doc": {
        "_index": "_index",
        "_id": "_id",
        "_source": {
          "sorted_array": [
            9,
            6,
            5,
            5,
            4,
            3,
            2,
            1,
            1
          ],
          "my_array_field": [
            3,
            1,
            4,
            1,
            5,
            9,
            2,
            6,
            5
          ]
        },
        "_ingest": {
          "timestamp": "2024-05-30T22:10:13.405692128Z"
        }
      }
    }
  ]
}
```
{% include copy-curl.html %}

### Step 3: Ingest a document 

The following query ingests a document into an index named `testindex1`:

```json
POST testindex1/_doc?pipeline=sort-pipeline
{
  "my_array_field": [3, 1, 4, 1, 5, 9, 2, 6, 5]
}
```
{% include copy-curl.html %}

#### Response

> The request indexes the document into the index `testindex1` and then indexes and then indexes all documents with the `my_array_field` sorted in descending order, as shown in the following response:

Reviewer: ```suggestion
The request indexes the document into the index `testindex1` and then indexes all documents with the `my_array_field` sorted in descending order, as shown in the following response:
```
